src/Core/Services/CipherService.cs

```diff
@@ -556,20 +556,9 @@ public async Task ShareWithServerAsync(CipherView cipher, string organizationId,
 
         public async Task<Cipher> SaveAttachmentRawWithServerAsync(Cipher cipher, string filename, byte[] data)
         {
-            SymmetricCryptoKey attachmentKey;
-            EncString protectedAttachmentKey;
-            var orgKey = await _cryptoService.GetOrgKeyAsync(cipher.OrganizationId);
-            if (orgKey != null)
-            {
-                (attachmentKey, protectedAttachmentKey) = await _cryptoService.MakeDataEncKeyAsync(orgKey);
-            }
-            else
-            {
-                var userKey = await _cryptoService.GetUserKeyWithLegacySupportAsync();
-                (attachmentKey, protectedAttachmentKey) = await _cryptoService.MakeDataEncKeyAsync(userKey);
-            }
+            var (attachmentKey, protectedAttachmentKey, encKey) = await MakeAttachmentKeyAsync(cipher.OrganizationId);
 
-            var encFileName = await _cryptoService.EncryptAsync(filename, orgKey);
+            var encFileName = await _cryptoService.EncryptAsync(filename, encKey);
```

Comment:
🤔 Just in case, is it ok to pass `encKey` as parameter to encrypt the filename? Because in the previous logic the `orgKey` was being passed so if it was `null` then it was passing `null` as parameter but with the `encKey` now if the `orgKey` is `null` the `MakeAttachmentKeyAsync` will return the `userKey`, so the parameter will end ups being the `userKey` for the filename encryption. Is that the right logic?
Btw, it also affects `ShareAttachmentWithServerAsync`

Reply:
Should be okay here. We're using `GetUserKeyWithLegacySupportAsync` anyways inside that method if the key provided is `null`